llarp/service/lns_tracker.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <functional>
+#include <optional>
+#include <unordered_map>
+#include <unordered_set>
+#include <string>
+
+#include "address.hpp"
+
+namespace llarp::service
+{
+  /// tracks and manages consensus of lns names we fetch from the network
+  class LNSLookupTracker
+  {
+    struct LookupInfo
+    {
+      std::unordered_set<Address> m_CurrentValues;
+      std::function<void(std::optional<Address>)> m_HandleResult;
+      std::size_t m_ResultsGotten = 0;
+      std::size_t m_ResultsNeeded;
+
+      LookupInfo(std::size_t wantResults, std::function<void(std::optional<Address>)> resultHandler)
+          : m_HandleResult{std::move(resultHandler)}, m_ResultsNeeded{wantResults}
+      {}
+
+      /// return true to remove self
```

Comment:
that's a very un-intuitive use of a boolean return.

perhaps add a function "Done" or similar, then above when using do

`if (itr = find):
    itr->HandleOneResult;
    if itr->Done():
        erase(itr)`

Reply:
fixed